lib/rules/window-is-top-matches.js

```diff
@@ -1,5 +1,8 @@
+import cache from '../core/base/cache';
+
 function windowIsTopMatches(node) {
 	return (
+		cache.get('isTopWindow') ||
```

Comment:
Why not pass the options around, or record what we think is the top level in the context object and pass that around? This is basically using the cache as a way to create global variables. Global variables is something we're trying to get out of, lets not add more of them.

Reply:
I agree, but think this might be a naming problem. If `cache` was called `globalState`, I don't think we'd have any issue putting "global data" here.

Reply:
The main issue is that we pass both the `context` and the `options` directly to the iframes, so any property we put into either of those will also trigger in the iframes.

Reply:
Just say that `initiator` variable on the context. Instead of an options, we could just always run those rules if `initiator: true` is set (which is always the top-most `run` call).

Reply:
Deprecated `window-is-top` matches for `is-initiator-matches` and just check if the context is the initiator to run those rules.